### DISCUSSION TITLE
ubl 10 x 10 only 65 probes

### DISCUSSION THREAD

**Discussion by ghost** [Jyers/Marlin#520] [Q&A] [answered]
Why is my creality board only doing 65 times of probing on my bed instead of 100 when i do the ubl leveling ?
Does anyone has the same problems ? 
I loaded up the E3V2-UBL-BLTouch-10x10-v4.2.7.bin for my board and getting some weird troubles.
 im getting alot of thermal runaways on my bed heating as well 
when the print started after a minute or so the printer starts to beep i have this over and over again.
Does the creality board 4.2.7 has less memory or do i have a faulty board ?
staring to think to buy the skr mini e3 v2 2.1 board only the thing is does this board run with the stock lcd screen from creality

**Comment by ghost** [comment]
The same happened to me.
I am on the 4.2.2

**Reply by ghost** [reply]
i don't understand why this is happening if i flash the board to stock creality everything is fine 
when i use this firmware it's not working at all

**Reply by ghost** [reply]
I have some serious issues with BLTouch, not sure what is going on the level is too high on one side and too low on the other side of the bed, I might try to use the reality software again soon.

**Reply by ghost** [reply]
i did ordered a new 4.2.7 board from creality to check if i have a faulty board 
and i ordered a skr mini e3 2.1, i probably going to change it in the future on this board there more options than on the stock board.
i think that the eeprom cant handle to much data when i use ubl probing 
the board can only save 2 meshes and its only probing 65 times.
the new creality board wil arrive within 10 days so im very curious if my board is faulty or the firmware has some bugs

**Comment by G4GNX** [comment]
I saw that with the 15x15 but I put it down to the fact that the probe can't get to every part of the bed, so the UBL firmware does what it's supposed to do, fills in the gaps.
I also believe it's a spiral probe, not a square probe, which is why the pattern is not symmetrical.

**Comment by Jyers** [comment, marked as answer]
This is the correct behavior, see #403

**Reply by ghost** [reply]
ok i understand but now the therminal runaway with stock firmware from creality no problems with this firmware if have within a minute a therminal  on my bed what i cant explain

**Reply by Jyers** [reply]
Maybe try PID tuning.

**Reply by ghost** [reply]
i did that nothing changed i did the pid tuning with prontface and on the display 
problem stayed the same 
i had to order a new board let see if this problem that i have stays or solve by dropping a new board in

**Reply by Jyers** [reply]
Okay, well keep me updated if the problem persists.

**Reply by ghost** [reply]
will do thanks for the help for so far

**Reply by ghost** [reply]
> This is the correct behavior, see #403

Hi Jyers, Creality is sending me a new mainboard this one is faulty
keep you posted when it arrives and put back in to the machine

**Reply by njmV** [reply]
I don't know if this is a nice solution?, but in my case I managed to recalc some values that helped me to have my 10x10 grid with all 100 spots

In summary on **Configuration_adv.h** I adjusted

  _#define MESH_MIN_X MESH_INSET+**46**
  #define MESH_MIN_Y MESH_INSET+**8**
  #define MESH_MAX_X X_BED_SIZE - (MESH_INSET+**46**)
  #define MESH_MAX_Y Y_BED_SIZE - (MESH_INSET+**8**)_

The -46 and -8 are the values from my **Configuration.h** for NOZZLE_TO_PROBE_OFFSET { -46, -8, 0 }

I also kept PROBING_MARGIN>MESH_INSET. In summary I adjusted:
  _#define MESH_INSET 5
  #define PROBING_MARGIN 7_

**Reply by Jyers** [reply]
@njmV I'm going to reiterate what I've said a lot about this. While it might seem ideal to probe all 100 points, it's not beneficial in the long run as you will then have areas of your bed that don't have mesh values and therefore will not be compensated. If you want a more detailed mesh, leave the mesh covering the whole bed, increase the grid size and let the firmware infer the unreachable points

**Reply by njmV** [reply]
@Jyers: thank you for the time! I greatly appreciate.
I understand what you shared.

I also doubt about my own configuration because I do not understand how the grid spots are evaluated.
I think things might work like this, but completely guessing...

If I have a 10x10 (100) spots in a grid I can mechanically get all spots from grid except (in my case) a little right blind margin of 25mm. (by right I mean looking at the printer from front)

So, mesh spots after 230-25=205mm cannot be physically measured.

My bed is 230mm, then removing PROBING_MARGIN x 2 I think I will get 230-7x2=216mm for probing. 
Dividing 216/10, each spot measuring distance will be +-21,6mm

Starting @PROBING_MARGIN (mine is 7)
I think the 10 probing for one row will be similar to 7, 28.6, 50.2, 71.8, 93.4, 115, 136.6, 158.2, 179.8, 201.4 – meaning I could probe all 10 points, losing the right 25mm (spots above 205mm).

My Y is physically measurable for all 230mm, so I think I could measure 10x10, with a great grid coverage? 

**PS: I am not sure that my thoughts are correct, just wondering 😊!**

PS2: After reading again your post and MESH_MAX_* parameters I think I understand what it does is to reduce the mesh size and not, (as I thought) adjusting the bltouch "correctly". sorry.

**Reply by Jyers** [reply]
Yeah, that seems like a good methodology. It's all about how much print space you're gonna use, as if you only print where your BL Touch can reach anyway it doesn't matter, but it your going to print all the way to the edges you need to have mesh points out there.
